pkg/storage/batch.go

```diff
@@ -139,6 +139,36 @@ func (r *BatchReader) Value() []byte {
 	}
 }
 
+// Verify verifies the checksums of the current record in the batch entry
+// will return error on mismatch.
+func (r *BatchReader) Verify() error {
+	key, err := r.EngineKey()
+	if err != nil {
+		return err
+	}
+	// TODO: verify values with other encoding types.
```

Comment:
Do we have an idea which types are missing? My understanding is that the checksums we're checking are part of `MVCCValue`. So it seems like we're covering it all?

nit: TODO should be named like TODO(lyang24), or reference an issue.

Reply:
Looking at https://github.com/cockroachdb/cockroach/blob/4e4644352bb6bfdd7b4045e2a5dbbb89d3f68dfa/pkg/storage/verifying_iterator.go#L40-L50, is it true that we only include a checksum for point keys? Is this because range keys don't have any value? It's possible that we don't need this TODO here, and can just mirror what `verifyingMVCCIterator` does.

@erikgrinaker

Reply:
> Do we have an idea which types are missing? My understanding is that the checksums we're checking are part of MVCCValue. So it seems like we're covering it all?

The checksums are in `roachpb.Value`, not `MVCCValue`. These can also be found elsewhere. See my other comment about `MVCCMetadata.RawBytes`.